master/internal/api_experiment.go

```diff
@@ -282,18 +283,15 @@ func (a *apiServer) DeleteExperiment(
 	return &apiv1.DeleteExperimentResponse{}, nil
 }
 
-func (a *apiServer) deleteExperiment(exp *model.Experiment, user *model.User) error {
+func (a *apiServer) deleteExperiment(exp *model.Experiment, userModel *model.User) error {
 	conf, err := a.m.db.LegacyExperimentConfigByID(exp.ID)
 	if err != nil {
 		return fmt.Errorf("failed to read config for experiment: %w", err)
 	}
 
-	agentUserGroup, err := a.m.db.AgentUserGroup(*exp.OwnerID)
-	switch {
-	case err != nil:
-		return errors.Errorf("cannot find user and group for experiment")
-	case agentUserGroup == nil:
-		agentUserGroup = &a.m.config.Security.DefaultTask
+	agentUserGroup, err := user.GetAgentUserGroup(*exp.OwnerID, exp)
```

Comment:
that is a good point. with the secure storage setup, I'd expect the system ideally should also move the underlying storage as well (because it may be in a different bucket, different fs etc.). But we aren't setup for this today, so a user would need to manually do this data move.
Otherwise their experiments within determined and in storage diverge in terms of security.